CMakeLists.txt

```diff
@@ -209,16 +209,7 @@ endif(ENABLE_ORC)
 ########################################################################
 find_package(Doxygen)
 if(DOXYGEN_FOUND)
-    configure_file(
-        ${CMAKE_SOURCE_DIR}/Doxyfile.in
-        ${CMAKE_BINARY_DIR}/Doxyfile
-    @ONLY)
-
-    add_custom_target(volk_doc
-        ${DOXYGEN_EXECUTABLE} ${CMAKE_BINARY_DIR}/Doxyfile
-        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-        COMMENT "Generating documentation with Doxygen" VERBATIM
-    )
+  add_subdirectory(docs)
```

Comment:
I need to figure out how to edit PRs ... regardless: Let's move this whole section, from `find_package(Doxygen)` to `endif(DOXYGEN_FOUND)` into the subdirectory. That's better CMake programming: only check in the current CMakeLists.txt file for stuff that's required within it, and don't rely on previous checks that may or not have taken place.

Reply:
I agree. Are there actually any resources that teach you "good CMake"?

Reply:
Hmmm ... good question ... IDK .. mine is via experience, having used CMake now since GR switched, ages ago; and, working on tons or projects that use CMake

Reply:
I moved it.